src/lib/array.jl

```diff
@@ -41,6 +41,15 @@ end
   return (_project(x, dx), map(_->nothing, inds)...)
 end
 
+using CUDA
+import NNlib
+
+∇getindex(x::CuArray, inds::Tuple{AbstractArray{<:Integer}}) = dy -> begin
+  dx = _zero(x, eltype(dy))
+  NNlib.scatter!(+, dx, dy, inds[1])
```

Comment:
since we use only `inds[1]` we should restict the input to `::AbstractArray{<:Integer}` or even better `AbstractVector{<:Integer}`

Reply:
The tuple wrapper comes from the varargs in the adjoint, so unless that's changed too there's not much that can be done.